init.hh

```diff
@@ -60,10 +60,29 @@ public:
         return initialize(map);
     }
 
+    enum class Event {
+        Started,
+        Stopped,
+    };
```

Comment:
We use snake_case for things, including enums.

What does it mean for a configurable to be started or stopped? I understand changed, but how can you start a config item?

Reply:
Is it for the configuration mechanism itself?

Reply:
It is for the overall system, i.e. initiator. A "service" creates a configurable listener, which is global, but stateless. A main/init/whatever calls this set of "hooks" with config etc for this init sequence, _before_ init really starts. The service can then augment things like config and extensions, adding stateful objects to this, that are local to the actual init sequence. 
These callbacks are optional notifications that can be received once this init sequence reached certain points, i.e. "somewhat done" (all things up but not yet accepting user requests") and "stopped" which mean shutdown is more or less complete, i.e. hooked objects can do some cleanup if they want.

Reply:
Snakes removed